source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -276,7 +280,8 @@ def getPods(namespace)
     def getWindowsNodes
       winNodes = []
       begin
-        resourceUri = getNodesResourceUri("nodes")
+        # get only windows nodes
+        resourceUri = getNodesResourceUri("nodes?labelSelector=kubernetes.io%2Fos%3Dwindows")
```

Comment:
what happens for older nodes with beta label? Its not good to hardcode label selectors in code as labels are highly volatile.

Reply:
This standard node label and which is available starting from 1.14 https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.14.md#deprecations and verified the same.  From our telemetry, we have 2 clusters under < 1.14, out of this one already failed provisioning state and other one doesnt have any windows nodes from AKS telemetry. From k8s doc, windows container supported added in 1.14 https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/ so I think, we are good on this.